trackintel/analysis/location_identification.py

```diff
@@ -301,24 +301,35 @@ def osna_method(staypoints):
 
     # create a pivot table -> labels "home" and "work" as columns. ("user_id", "location_id" still in index.)
     sp_pivot = sp_agg.unstack()
-    # this line is here to circumvent a bug in pandas .idxmax should return NA if all values in column are
-    # empty but that doesn't work for pd.NaT -> therefor we cast to float64
-    sp_pivot /= pd.Timedelta("1ns")
     # get index of maximum for columns "work" and "home"
     # looks over locations to find maximum for columns
-    sp_idxmax = sp_pivot.groupby(["user_id"]).idxmax(axis="index")
-    # first assign labels
-    for col in sp_idxmax.columns:
-        sp_pivot.loc[sp_idxmax[col].dropna(), "purpose"] = col
-
-    # The "home" label could overlap with the "work" label
-    # we set the rows where "home" is maximum to zero (pd.NaT) and recalculate index of work maximum.
-    if all(col in sp_idxmax.columns for col in ["work", "home"]):
-        redo_work = sp_idxmax[sp_idxmax["home"] == sp_idxmax["work"]]
-        sp_pivot.loc[redo_work["work"], "purpose"] = "home"
-        sp_pivot.loc[redo_work["work"], "work"] = np.nan
-        sp_idxmax_work = sp_pivot.groupby(["user_id"])["work"].idxmax()
-        sp_pivot.loc[sp_idxmax_work.dropna(), "purpose"] = "work"
+    # use fillna to such that idxmax works on columns with only NaT
+    sp_idxmax = sp_pivot.fillna(pd.Timedelta(0)).groupby(["user_id"]).idxmax()
```

Comment:
Is the `fillna` necessary? What is the behavior when we do not add it? 
Or in other words, in the original comments, you state that there is a bug in pandas. Does it still exist?

Reply:
In future releases this line would raise a `ValueError`, this is also true for the bug mentioned.
They have changed this behavior to be consistent with `np.nanargmax` which is called in `idxmax`.
So the bug is fixed, but for that we need to make sure that no full-NaT groups exist.

Reply:
Just to clarify: what do you mean by `In future releases`? 
I think the ValueError in `np.nanargmax` is what you are referring to, is this correct?

Reply:
What I mean is that currently only a future warning is issued because the behaviour is deprecated. But in future releases this will change.
Yes exactly, for a pandas Series with only `np.nan` the ValueError in `np.nanargmax` does not trigger, which leads to this inconsistent behaviour.
Therefore, they will check and raise the error themselves.